### DISCUSSION TITLE
[info] Compiles fine in Windows

### DISCUSSION THREAD

**Discussion by 59de44955ebd** [tsujan/FeatherPad#818] [Show and tell]
Hi,
I just wanted to let you know that - just for fun - today I compiled FeatherPad for Windows (using MSYS2), and there were hardly any issues. Attached screenshot shows it running in Windows 11.

Apart from some minor UI stuff - e.g. menubar and statusbar could be adjusted a little to look better in Windows - there is only a single file that didn't compile out of the box: "signalDaemon.cpp", which depends on BSD sockets that are obviously not available in Windows. For now I just deactivated its functionality completely, but I guess it would be no big deal to make it Windows-compatible using C macros and WinSocket.

![featherpad_win11](https://github.com/user-attachments/assets/79a03ea1-6372-4d62-9aef-3a5a012ec3b8)

**Comment by 59de44955ebd** [comment]
ps: some technical details, this Windows version is based on Qt 6.7, and the plain binary (.exe) has a file size of 2,76 MB.

**Reply by ghost** [reply]
Can you build it with Qt5?

**Reply by 59de44955ebd** [reply]
My build was based on the master branch which seems to depend on Qt 6. But whatever it takes to compile FeatherPad with Qt5 instead of Qt6 (Other branch? Makefile switches?), this will propably also work when compiling it for Windows. MSYS2 supports both Qt5 and Qt6.

**Reply by tsujan** [reply]
FeatherPad said goodbye to Qt5 in its version [1.5.0](https://github.com/tsujan/FeatherPad/releases/tag/V1.5.0).

**Comment by tsujan** [comment]
Thanks a lot for the info!

I wonder if there was no problem about finding Hunspell.

If you complete the Window support and want to maintain it, please make a PR.

Please also note that
 * I only have Linux; and
 * No binaries are added here. So, if you want to add binaries, you could make a fork.

**Reply by 59de44955ebd** [reply]
I have to admit that I havn't tested localisation and spell checking yet, but the binary is linked (dynamically) to hunspell, and at runtime loads "libhunspell-1.7-0.dll" just fine. I'll test tomorrow if spell checking actually works and will then report back.

**Reply by tsujan** [reply]
> I havn't tested localisation and spell checking yet

Take your time.

> there is only a single file that didn't compile out of the box: "signalDaemon.cpp"

You could exclude `signalDaemon` from Windows if there isn't a simple solution there.

Judging by the works of the contributors who have added support for other OS's, I think a few OS checks may be needed for Windows inside the code.

**Reply by 59de44955ebd** [reply]
FYI, in the meantime I tested it, Hunspell works fine in this Windows version. It requires UTF-8 formated .dic files (like those at https://github.com/wooorm/dictionaries), but I guess that applies to any version of FeatherPad.

In the meantime I also forked the repo and added a release containing Windows binaries (as 7-zip archive and as installer):
https://github.com/59de44955ebd/FeatherPad/releases/tag/v1.0

**Comment by ghost** [comment]
Please have a look if you can learn something from @brechtsanders' patch:

https://github.com/brechtsanders/winlibs_recipes/blob/main/recipes/featherpad.winlib

I found it on https://github.com/tsujan/FeatherPad/issues/718.
